### PR TITLE
Shishkina Valeriya. Lab #1. Var #1.

### PR DESCRIPTION
**Pull request by ShLera04**
As a result of the laboratory work, a plugin was written that outputs the names of all classes and their fields.
For example:
```C++
struct test1
{
    int a;
    int b;
};
```
Input:
```
test1
  |_ a
  |_ b
```

### CODE REVIEW

File: clang/labs/lab1/shishkina_valeriya/PrintClasses.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+public:
+  std::unique_ptr<clang::ASTConsumer> CreateASTConsumer (
+    clang::CompilerInstance &Compiler, llvm::StringRef InFile) override {
+      return std::make_unique<PrintClassesConsumer>(&Compiler.getASTContext());
+    }
+
+protected:
+  bool ParseArgs(const clang::CompilerInstance &Compiler, 
+                 const std::vector<std::string> &args) override {
+    return true;
```
Comment (m-ly4) [inline clang/labs/lab1/shishkina_valeriya/PrintClasses.cpp:60]: Please add a `help` argument and a test for it.

File: clang/labs/lab1/shishkina_valeriya/PrintClasses.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+
+class PrintClassesVisitor : public clang::RecursiveASTVisitor<PrintClassesVisitor> {
```
Comment (m-ly4) [inline clang/labs/lab1/shishkina_valeriya/PrintClasses.cpp]: Please fix diff suggestions of clang-format test: https://github.com/NN-complr-tech/llvm-nnsu-2024/actions/runs/8310037939/job/22742071220?pr=18#step:4:14

File: clang/labs/lab1/shishkina_valeriya/PrintClasses.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+class PrintClassesPlugin : public clang::PluginASTAction {
+public:
+  std::unique_ptr<clang::ASTConsumer> CreateASTConsumer (
+    clang::CompilerInstance &Compiler, llvm::StringRef InFile) override {
+      return std::make_unique<PrintClassesConsumer>(&Compiler.getASTContext());
+    }
+
+protected:
+  bool ParseArgs(const clang::CompilerInstance &Compiler, 
+                 const std::vector<std::string> &args) override {
```
Comment (m-ly4) [inline clang/labs/lab1/shishkina_valeriya/PrintClasses.cpp]: Could you please fix clang-tidy warnings if applicable: https://github.com/NN-complr-tech/llvm-nnsu-2024/actions/runs/8310037939/job/22742071140?pr=18#step:6:489

**Review by m-ly4 — CHANGES_REQUESTED** [review]

File: clang/test/lab1/shishkina_valeriya/tests.cpp [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+{
+    // CHECK-NEXT: |_ Arg
+    static int Arg;
+public:
+    // CHECK-NEXT: |_ Brr
+    int Brr = 2;
+};
+
+// CHECK: Test4
+struct Test4{};
```
Comment (m-ly4) [inline clang/test/lab1/shishkina_valeriya/tests.cpp]: Please add a test for `help` argument.

**Review by aobolensk — APPROVED** [review]

**Comment by m-ly4** [comment]
@ShLera04 Arseniy have already approved the PR. But anyway could you please fix the test?

**Review by m-ly4 — APPROVED** [review]

**Pull request merged**